lib/utils.js

```diff
@@ -7,98 +10,79 @@ var assert = require("assert");
  * @returns Configuration Object
  */
 exports.parseMongoConnectionURL = function (mongoConnectionURL) {
-  var result = {};
 
-  //The connection string will be of the form mongodb://user:password@host:port,host2port2/databasename?option=someOption
+  //The connection string will be of the form mongodb://username:pass@db1.prv.mbaas2.cluster.feedhenry.net,db2.prv.mbaas2.cluster.feedhenry.net:27017/domain-vvuq7douloe2zkyktub7fcmt-demos-dev?replicaSet=cluster-mbaas2_rs1
   //String to break down the mongodb url into a ditch config hash
-  var auth_hosts_path_options = new Buffer(mongoConnectionURL).toString().split("//")[1];
+  const parts = mongoConnectionURL.split(',');
 
-  //user:password , host:port,host2:port2/databasename?option=someOption
-  var auth = auth_hosts_path_options.split("@")[0];
+  // The name comes from the path in the URL. We must always use the last host
+  function getName () {
+    const pathname = url.parse(parts[parts.length - 1]).pathname;
 
-
-  var hosts_path_options = auth_hosts_path_options.split("@")[1];
-
-  // host:port,host2:port2/databasename , option=someOption
-  var hosts_path = hosts_path_options.split("?")[0];
-  var options = hosts_path_options.split("?")[1];
-
-  //host:port,host2:port2, databasename
-  var hosts = hosts_path.split("/")[0];
-  var databaseName = hosts_path.split("/")[1];
-
-  //host:port host2:port2
-  var hostList = hosts.split(",");
-
-  //user, password
-  var authUser = auth.split(":")[0];
-  var authPassword = auth.split(":")[1];
-
-  result.database = {};
-  result.database.driver_options = {};
-  result.database.auth = {};
-  result.database.auth.user = authUser;
-  result.database.auth.pass = authPassword;
-  result.database.auth.source = databaseName;
-  result.database.name = databaseName;
-
-  assert.ok(authUser !== undefined);
-  assert.ok(authPassword !== undefined);
-  assert.ok(databaseName !== undefined);
-
-  //Parsing options
-  if (options) {
-    var parsedOptions = options.split(",");
-    //If options are parsed, they should have an even number of elements in the split array
-    assert.ok(parsedOptions.length > 0);
-
-    for (var i = 0; i < parsedOptions.length; i++) {
-      var splitOption = parsedOptions[i].split("=");
-      //Each option should be something=value
-      assert.ok(splitOption && splitOption.length === 2);
-      assert.ok(splitOption[0].length > 0);
-      assert.ok(splitOption[1].length > 0);
-      //Checking it does not already exist
-      assert.ok(!result.database.driver_options[splitOption[0]]);
-
-      result.database.driver_options[splitOption[0]] = splitOption[1];
-    }
+    return pathname ? pathname.replace('/', '') : pathname;
```

Comment:
Is not required send the options? Is the replica set working without the options?